frontend/src/features/toy/ToyList.tsx

```diff
@@ -5,67 +5,100 @@ import { RootState } from "../../store/store";
 import { fetchToys, setPage, setResultsPerPage } from "./toySlice";
 import ToyCard from "./ToyCard";
 import Pagination from "../../components/Pagination";
+import Error from "../../components/Error";
+import Loader from "../../components/Loader";
+import ToyFilterList from "./Filter/ToyFilterList";
 
 function ToyList() {
```

Comment:
Optimize the dispatch logic and reduce redundancy.
`function ToyList() {
  const dispatch = useAppDispatch();
  const { toys, error, loading, currentPage, totalPages, resultsPerPage, totalResults, ageCategory, brand, category, sort, search } = useAppSelector(state => state.toys);

  useEffect(() => {
    const fetchToysData = () => {
      const params = {
        page: currentPage,
        limit: resultsPerPage,
        category: category.join(","),
        ageCategory: ageCategory.join(","),
        brand: brand.join(","),
        sort,
        search
      };
      dispatch(fetchToys(params));
    };

    fetchToysData();
  }, [dispatch, currentPage, resultsPerPage, category, ageCategory, brand, sort, search]);

  if (loading) return <Loader />;
  if (error) return <Error errorMessage={error ?? "Something went wrong.."} />;

  return (
    <div className="min-h-[100%] relative">
      <section className="flex gap-5 px-4 sm:px-10">
        <ToyFilterList />
        <section>
          <Pagination
            currentPage={currentPage}
            totalPages={totalPages}
            onPageChange={page => dispatch(setPage(page))}
            resultsPerPage={resultsPerPage}
            onResultsPerPageChange={results => {
              dispatch(setPage(1));
              dispatch(setResultsPerPage(results));
            }}
            totalResults={totalResults}
          />
          <div className="flex flex-wrap justify-center ">
            {toys.map(toy => (
              <ToyCard
                key={toy._id}
                id={toy._id}
                name={toy.name}
                description={toy.description}
                images={toy.images.length > 0 ? toy.images : ["../bear.webp"]}
                tokens={toy.tokenValue}
              />
            ))}
          </div>
          <Pagination
            currentPage={currentPage}
            totalPages={totalPages}
            onPageChange={page => dispatch(setPage(page))}
            resultsPerPage={resultsPerPage}
            onResultsPerPageChange={results => dispatch(setResultsPerPage(results))}
            totalResults={totalResults}
          />
        </section>
      </section>
    </div>
  );
}
`